docs/architecture.rst

```diff
@@ -129,44 +129,64 @@ a lookup is done against these ``chids``.
 
 .. _table-rotation:
 
-Message Table Rotation
-----------------------
+Message Table Rotation (legacy)
+-------------------------------
+
+As of version 1.45.0, message table rotation can be disabled. This is because
+DynamoDB now provides automatic entry expiration. This is controlled in our
+data by the "expiry" field. (***Note***, field expiration is only available in
+full DynamoDB, and is not replicated with the mock DynamoDB API provided for
+development.) The following feature is disabled with the `no_table_rotation`
+flag set in the `autopush_shared.ini` configuration file.
+
+If table rotation is disabled, the last message table used will become
+'frozen' and will be used for all future messages. While this may not be
+aesthetically pleasing, it's more efficient than copying data to a new,
+generic table. If it's preferred, service can be shut down, previous tables
+dropped, the current table renamed, and service brought up again.
 
-To avoid costly table scans, autopush uses a rotating message and router table.
-Clients that haven't connected in 30-60 days will have their router and message
-table entries dropped and need to re-register.
+..
 
-Tables are post-fixed with the year/month they are meant for, ie:
+  *Message Table Rotation information*
 
-    messages-2015-02
+  To avoid costly table scans, autopush uses a rotating message and router
+  table.
+  Clients that haven't connected in 30-60 days will have their router and
+  message table entries dropped and need to re-register.
 
-Tables must be created and have their read/write units properly allocated by a
-separate process in advance of the month switch-over as autopush nodes will
-assume the tables already exist. Scripts are provided that can be run weekly to
-ensure all necessary tables are present, and tables old enough are dropped.
+  Tables are post-fixed with the year/month they are meant for, i.e. ::
 
-.. seealso::
+     messages_2015-02
```

Comment:
missed the last dash, should be 2015_02, heh. I think this needs a rebase now, anyway